pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -233,3 +233,42 @@ load 1m
 
 eval range from 0 to 7m step 1m -metric
   {} -1 -1 -1 -3 -3 _ -4 -4
+
+clear
+
+# label_replace() tests
```

Comment:
Could you please add a test for the case where different series should merge into one after relabelling? Something like this:

```
load 1m
  series{label="a", idx="1"} 2 _
  series{label="a", idx="2"} _ 4

eval range from 0 to 1m step 1m label_replace(series, "idx", "replaced", "idx", ".*")
  series{label="a", idx="replaced"} 2 4
```

This should catch the missing `DeduplicateAndMerge`.

Reply:
This is fixed in this branch now, but has a related upstream problem https://github.com/prometheus/prometheus/issues/15114